demos/smart_classroom_demo_gapi/main.cpp

```diff
@@ -706,341 +175,232 @@ int main(int argc, char* argv[]) {
+                /** Main demo scenario **/
+                cv::GOpaque<FaceTrack> face_track;
+                cv::GOpaque<size_t> work_num_frames;
+                /** Recognize actions and faces **/
+                std::tie(tracked_actions, face_track, work_num_frames)
+                    = custom::GetRecognitionResult::on(in, faces, persons_with_actions, embeddings, frec_kernel_input, const_params);
+
+                cv::GOpaque<std::string> stream_log, stat_log, det_log;
+                cv::GArray<std::string> face_ids(face_id_to_label_map);
+                /** Get roi and labels for drawing and set logs **/
+                std::tie(draw_elements, stream_log, stat_log, det_log)
+                    = custom::RecognizeResultPostProc::on(in, tracked_actions, face_track, face_ids, work_num_frames, const_params);
+                /** Main demo part of graph output **/
+                outs += GOut(work_num_frames, stream_log, stat_log, det_log);
+            } else {
+                /** Top action case **/
+                cv::GMat top_k;
+                /** Recognize actions **/
+                tracked_actions =
+                    custom::GetActionTopHandsDetectionResult::on(in, persons_with_actions);
+                 /** Get roi and labels for drawing **/
+                std::tie(draw_elements, top_k) = custom::RHands::on(in, tracked_actions, const_params);
+                /** Top action case part of graph output **/
+                outs += GOut(top_k);
+            }
+            /** Draw ROI and labels **/
+            auto rendered = cv::gapi::wip::draw::render3ch(frame,
+                                                           custom::BoxesAndLabels::on(frame, draw_elements, const_params));
+            /** Last graph output is frame to draw **/
+            outs += GOut(rendered);
+            return cv::GComputation(cv::GIn(in), std::move(outs));
+        });
+        auto networks = cv::gapi::networks(det_net, landm_net, reident_net, action_net);
+        cv::GStreamingCompiled cc = pp.compileStreaming(cv::compile_args(custom::kernels(),
+                                                        networks,
+                                                        TrackerParamsPack{ tracker_reid_params, tracker_action_params },
+                                                        logger_params));
+
+        /** The execution part **/
         std::cout << "To close the application, press 'CTRL+C' here";
         if (!FLAGS_no_show) {
             std::cout << " or switch to the output window and press ESC key";
         }
         std::cout << std::endl;
 
-        if (actions_type != TOP_K) {
-            action_detector->enqueue(frame);
-            action_detector->submitRequest();
-            face_detector->enqueue(frame);
-            face_detector->submitRequest();
-        }
+        cc.setSource(cv::gin(cv::gapi::wip::make_src<cv::gapi::wip::CustomCapSource>(cap)));
```

Comment:
Please use `setSource<>()` version since it is less scary (no `make_src` is required)